HabitTracker.harris-andy/Program.cs

```diff
@@ -0,0 +1,592 @@
+}
+
+int GetQuantityInput()
+{
+    int quantity = -1;
+    while (quantity < 0)
+    {
+        string message = "Enter amount of activity done/consumed/lost/whatever (greater than 0). Or press 0 to return to Main Menu";
+        quantity = validateNumberEntry(message);
+    }
+    return quantity;
+}
+
+string GetHobby()
+{
+    string? hobby = "";
+    while (string.IsNullOrWhiteSpace(hobby))
+    {
+        Console.WriteLine("Enter name of activity (keep it short). Or press 0 to return to Main Menu");
+        string? temp = Console.ReadLine();
+        hobby = temp?.Trim().ToLower();
+    }
+    if (hobby == "0") MainMenu();
+    return hobby;
+}
+
+void GetAllRecords(bool withIds)
+{
+    Console.Clear();
+    string sortMessage = "Sort records by\n1. ID\n2. Date";
+    int sortOption = -1;
+    while (sortOption < 0 || sortOption > 2)
+    {
+        sortOption = validateNumberEntry(sortMessage);
+    }
+
+    using (SqliteConnection connection = new SqliteConnection(connectionString))
+    {
+        List<HobbyRecord> hobbiesSortedID = new List<HobbyRecord>();
+
+        connection.Open();
+        using (SqliteCommand command = new SqliteCommand("SELECT * FROM habits", connection))
+        {
+            using (SqliteDataReader reader = command.ExecuteReader())
+            {
+                while (reader.Read())
+                {
+                    hobbiesSortedID.Add(new HobbyRecord
+                    {
+                        Id = reader.GetInt32(0),
+                        Date = DateTime.ParseExact(reader.GetString(1), format: "dd-MM-yyyy", new CultureInfo("en-US")),
+                        Hobby = reader.GetString(2),
+                        Units = reader.GetString(3),
+                        Quantity = reader.GetInt32(4)
+                    });
+                }
+            }
+        }
+        connection.Close();
+        Console.WriteLine("--------------------------------------------------\n");
+        Console.WriteLine("Here's all the fun stuff you did!\n");
+        if (hobbiesSortedID.Count == 0)
+        {
+            Console.WriteLine("No records found. Do stuff!");
+        }
+        var hobbiesSortedDate = hobbiesSortedID.OrderBy(record => record.Date).ToList();
+        foreach (HobbyRecord record in sortOption == 1 ? hobbiesSortedID : hobbiesSortedDate)
+        {
+            string outputByDate = $"{record.Date.ToString("dd-MMM-yyyy"),-13} {record.Hobby,-14} {record.Units,-5}: {record.Quantity,-5}";
+            string outputByID = $"{record.Id,-3}: {record.Date.ToString("dd-MMM-yyyy"),-13} {record.Hobby,-14} {record.Units,-5}: {record.Quantity,-5}";
+            if (sortOption == 1) Console.WriteLine(outputByID);
+            if (sortOption == 2) Console.WriteLine(outputByDate);
+        }
+        Console.WriteLine("--------------------------------------------------\n");
+    }
+}
+
+void GetRecordSummary()
+{
+    (string searchTerm, string searchTermCategory) = GetSearchTerm();

```

Comment:
Instead of using a tuple here this is calling out for some OOP. A SeachTerm class or similar.